File: src/polar_coordinates.cpp
//------------------------------------------------------------------------------
// polar_coordinates.cpp - содержит процедуру ввода параметров
// для уже созданных полярных координат
//------------------------------------------------------------------------------

#include "polar_coordinates.h"
#include <fstream>
#include <cmath>
//------------------------------------------------------------------------------
// Ввод полярных координат из файла
void In(polar_coordinates &pl, std::ifstream &ifst) {
    ifst >> pl.angel >> pl.x >> pl.y;
}

// Случайный ввод полярных координат
void InRnd(polar_coordinates &pl) {
    pl.angel = Random();
    pl.x = Random();
    pl.y = Random();
}

//------------------------------------------------------------------------------
// Вывод полярных координат в форматируемый поток
void Out(polar_coordinates &pl, std::ofstream &ofst) {
    ofst << "It is polar coordinates: angel = " << pl.angel << ", x coordinate = " << pl.x
            << ", y coordinate = " << pl.y
    <<". Convert to real number " << ConvertToRealNumber(pl) << "\n";
}

//------------------------------------------------------------------------------
// Вычисление действительного представления полярных координат
double ConvertToRealNumber(polar_coordinates &pl) {
    return std::sqrt(pl.x * pl.x + pl.y * pl.y);
}

File: hw 3/polar_coordinates.py
import math
from number import *


class PolarNumber(Number):
    def __init__(self):
        self.angle = 0.0
        self.x = 0
        self.y = 0

    def ReadStrArray(self, strArray, i):
        if i >= len(strArray) - 1:
            return 0
        self.x = int(strArray[i])
        self.y = int(strArray[i + 1])
        self.angle = int(strArray[i + 2])
        i += 3
        return i


    def Print(self):
        print("It is polar coordinates: angel = ", self.angle, ", x coordinate = ", self.x,
              ", y coordinate = ", self.y,
              ". Convert to real number ", self.ConvertToRealNumber())
        pass


    def Write(self, ostream):
        ostream.write("It is polar coordinates: angel = {}, x coordinate = {}, y coordinate = {}. Convert to real number {}"
                      .format(self.angle, self.x, self.y, self.ConvertToRealNumber()))
        pass


    def ConvertToRealNumber(self):
        return math.sqrt(self.x**2 + self.y**2)
        pass
